### ISSUE TITLE
Decompress files in parallel

### ISSUE THREAD

**Issue by vrmiguel** [ouch-org/ouch#244]
We can get big performance improvements by using a threadpool when decompressing files using stuff like Rayon.

[This commit](https://github.com/carmesim/ouch/commit/fe97ff813d2c3a4239f87108ab750b92adb86b59) is an example of how it can be somewhat easily integrable into ouch, but some quirks must be solved:

* error treatment must be adapted 
* threads try to print to stdout simultaneously creating garbled output

**Comment by marcospb19** [comment]
Solved by #337.

**Comment by figsoda** [comment]
This is still unimplemented for a lot of formats, but I guess the original issue was more about decompressing multiple archives

Here is a discussion on how we might be able to implement this for zip: https://github.com/zip-rs/zip/discussions/347#discussioncomment-4842053

Here is a related issue for tar: https://github.com/alexcrichton/tar-rs/issues/160

**Comment by figsoda** [comment]
I guess we can just open a new issue
